From 68a4f9e637793c2e27447d9bb5b3b22eeb0d64ad Mon Sep 17 00:00:00 2001
From: Lukas Czerner <lczerner@redhat.com>
Date: Wed, 29 Oct 2014 16:34:35 -0400
Subject: [fs] ext4: fix return value of ext4_do_update_inode

Message-id: <1414600485-1929-46-git-send-email-lczerner@redhat.com>
Patchwork-id: 98561
O-Subject: [RHEL 7.1 PATCH v2 45/55] ext4: fix return value of ext4_do_update_inode
Bugzilla: 1156096
RH-Acked-by: Eric Sandeen <sandeen@redhat.com>

From: Li Xi <pkuelelixi@gmail.com>

BZ 1156096
https://bugzilla.redhat.com/show_bug.cgi?id=1156096

BREW 8169722
https://brewweb.devel.redhat.com/taskinfo?taskID=8169722

Upstream commit bce92d566a57893e98ec83e4e5447f860d2889b7

 When ext4_do_update_inode() gets error from ext4_inode_blocks_set(),
 error number should be returned.

 Signed-off-by: Li Xi <lixi@ddn.com>
 Signed-off-by: Theodore Ts'o <tytso@mit.edu>
 Reviewed-by: Eric Sandeen <sandeen@redhat.com>
 Reviewed-by: Jan Kara <jack@suse.cz>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/ext4/inode.c b/fs/ext4/inode.c
index 2bb027f..742d714 100644
--- a/fs/ext4/inode.c
+++ b/fs/ext4/inode.c
@@ -4244,7 +4244,8 @@ static int ext4_do_update_inode(handle_t *handle,
  EXT4_INODE_SET_XTIME(i_atime, inode, raw_inode);
  EXT4_EINODE_SET_XTIME(i_crtime, ei, raw_inode);
 
- if (ext4_inode_blocks_set(handle, raw_inode, ei)) {
+ err = ext4_inode_blocks_set(handle, raw_inode, ei);
+ if (err) {
   spin_unlock(&ei->i_raw_lock);
   goto out_brelse;
  }
-- 
1.7.1